server/svix-server/src/expired_message_cleaner.rs

```diff
@@ -0,0 +1,44 @@
+// SPDX-FileCopyrightText: © 2022 Svix Authors
+// SPDX-License-Identifier: MIT
+
+use crate::db::models::message;
+use crate::error::Result;
+use chrono::{Duration, Utc};
+use sea_orm::{
+    entity::prelude::*, sea_query::Expr, DatabaseConnection, DbErr, EntityTrait, UpdateResult,
+};
+use tokio::time::sleep;
+
+/// Nullifies the payload column for expired messages
+async fn clean_expired_messages(
+    pool: &DatabaseConnection,
+) -> std::result::Result<UpdateResult, DbErr> {
+    // this logic exists to gracefully migrate existing message attempt responses
+    message::Entity::update_many()
+        .col_expr(
+            message::Column::Expiration,
+            Expr::value(Utc::now() + Duration::days(90)),
+        )
+        .filter(message::Column::Expiration.is_null())
```

Comment:
You actually made expiration not nullable.
You also want to check whether `payload` is null just so it hits the index.

Reply:
`payload` is not nullable, but I think `expiration` can be null.

👍 on the payload condition so we can make use of that index (which would resolve my previous reply about adding an additional index specifically for the `expiration` column)

Reply:
```
     pub expiration: DateTimeWithTimeZone,
```
This is what I meant. It's not nullable in Rust.
Anyhow, it should also be not nullable in SQL so you can remove it.